### PR TITLE
Add delegation tab to node-gui

### PR DESCRIPTION
**Pull request by azarovh**
Screenshot 2024-01-24 at 13 01 35

### CODE REVIEW

File: common/src/address/mod.rs [hunk 1]
```diff
@@ -107,6 +107,23 @@ impl<T: Addressable> Address<T> {
...
+        let obj = self.decode_object(cfg)?;
+        let prefix_len = obj.address_prefix(cfg).len();
+
+        let result = if self.address.len() < prefix_len + 8 {
+            self.to_string()
+        } else {
+            // prefix + 4 first chars + ... + 4 last chars
+            format!(
+                "{}...{}",
+                self.address[0..prefix_len + 4].to_owned(),
```
Comment (TheQuantumPhysicist) [inline common/src/address/mod.rs:121]: I'm not sure how forgiving clippy will be about this... it's heavily frowned upon to take a slice from strings. The correct, type-safe way to do it is to ensure it's all ASCII, convert it to a `Vec<u8>` (or use `chars()`), crop it, then back to String. Now we know that addresses are strictly ASCII so what you did is safe, but static analyzers are not that smart.

**Review by TheQuantumPhysicist — APPROVED** [review]

File: common/src/address/mod.rs (hunk 1 above)
Comment (azarovh) [reply common/src/address/mod.rs:121]: clippy is fine with it

Comment (azarovh) [reply common/src/address/mod.rs:121]: added a test

Comment (TheQuantumPhysicist) [reply common/src/address/mod.rs:121]: Thanks

Comment (azarovh) [reply common/src/address/mod.rs:121]: I had to add the solution with bytes in another PR because clippy starts complaining there

Comment (TheQuantumPhysicist) [reply common/src/address/mod.rs:121]: Told you...

I don't understand how this is happening... this is the second time this is happening... it works at first but then it complains. Very confusing.

**Pull request merged**